proxy/proxy.go

```diff
@@ -1092,17 +1111,18 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}()
 
 	defer func() {
-		accessLogEnabled, ok := ctx.stateBag[accesslog.AccessLogEnabledKey].(bool)
+		accessLogEnabled, ok := ctx.stateBag[al.AccessLogEnabledKey].(*al.AccessLogFilter)
 
 		if !ok {
-			accessLogEnabled = !p.accessLogDisabled
+			accessLogEnabled = &al.AccessLogFilter{Enable: !p.accessLogDisabled, Prefixes: nil}
 		}
+		statusCode := lw.GetCode()
 
-		if accessLogEnabled {
+		if shouldLog(statusCode, *accessLogEnabled) {
```

Comment:
Is there a reason to not pass the `accessLogEnabled` as a pointer?

Reply:
No specific reason. My go-fu is weak :sweat_smile: